src/cljs/chronograph_web/pages/create_organization/events.cljs

```diff
@@ -0,0 +1,71 @@
+(ns chronograph-web.pages.create-organization.events
+  (:require [re-frame.core :as rf]
+            [chronograph-web.db :as db]
+            [chronograph-web.http :as http]
+            [chronograph-web.events.routing :as routing-events]
+            [ajax.core :as ajax]
+            [day8.re-frame.http-fx]
+            [clojure.spec.alpha :as s]))
+
+(def ^:private create-organization-uri "/api/organizations")
+
+(def ^:private root-path :create-organization)
+(def ^:private status-path [root-path :status])
+(defn- form-params-path [k] [root-path :form-params k])
+
+(defn- form-errors-path
+  ([] [root-path :form-errors])
+  ([k] (conj (form-errors-path) k)))
+
+(def ^:private field-spec
+  {:slug :organizations/slug
+   :name :organizations/name})
+
+(defn- update-form-error [db k v]
+  (let [spec (field-spec k)]
+    (cond
+      (nil? spec)
+      (assoc-in db (form-errors-path k) :missing-spec)
+
+      (not (s/valid? spec v))
+      (assoc-in db (form-errors-path k) :invalid)
+
+      :else
+      (update-in db (form-errors-path) #(dissoc % k)))))
+
+(rf/reg-event-db
+  ::create-organization-form-update
+  (fn [db [_ k v]]
+    (-> db
+        (assoc-in status-path :editing)
+        (assoc-in (form-params-path k) v)
+        (update-form-error k v))))
+
+(rf/reg-event-fx
+  ::create-organization-form-submit
+  (fn [{:keys [db]} _]
+    {:db (assoc-in db status-path :creating)
+     :http-xhrio (http/post create-organization-uri
+                            {:params {:name (get-in db (form-params-path :name))
+                                      :slug (get-in db (form-params-path :slug))}
+                             :on-success [::create-organization-succeeded]
+                             :on-failure [::create-organization-failed]})}))
+
+(defn- organization-url [slug]
+  (str "/organization/" slug))
+
+(rf/reg-event-fx
+  ::create-organization-succeeded
+  (fn [{:keys [db]} [_ {:keys [id slug] :as response}]]
+    {:fx [[:dispatch [::routing-events/set-page {:route-params {:slug slug}
+                                                 :handler :organization-show}]]
+          [:dispatch [::routing-events/set-token (organization-url slug)]]]
```

Comment:
Simply setting the token should suffice, since `pushy` will call the `set-page` event by itself. Also we have a custom effect for setting the token, so we can do away with the `:dispatch` indirection and use that effect directly.

Reply:
Will try it out and use only set-token if pushy automatically sets page.

Reply:
This is done. Pushy does it, but was unable to because we were trying to dispatch-sync in an event handler (that isn't allowed).